### ISSUE TITLE
CLI: Add request and response validation

### ISSUE THREAD

**Issue by rajdip-b** [keyshade-xyz/keyshade#555]
## Description

The CLI currently uses no validation at all. We would like the cli to use `schema` package to add validation to its modules. This will include input (request) and output (response) validation. The idea is, before making an API call, we would like to inspect the request body with the request schema. If it passes, we make the call. Similarly, when we get back a response from the API, we match the response with the response schema and print only those fields that matches the response type.

## Solution

We would like to implement these changes across the following modules. The changes would include updating each and every command of every module to include validation.

- [ ] Workspace
- [ ] Workspace Role
- [ ] Workspace Membership
- [ ] Project
- [ ] Environment
- [ ] Secret
- [ ] Variable
- [ ] Integration
- [ ] Event

**Comment by yetanotheraryan** [comment]
Hi @rajdip-b, Let me just clarify the problem statement here, we want to put in request and response schema for validations for all the API's in the mentioned modules list right? using any validation library like Joi for example?

Looking forward to contribute on the same.

**Comment by rajdip-b** [comment]
Hey, all the validation is already set up in `schema` package. You can find it under `packages/schema` folder. You would need to import the schemas from the `schema` package one by one (for every module) and then validate the request and responses in the CLI app you can find under `apps/cli`. Hope that helps. 

You can also refer how `packages/api-client` uses the `schema` package.

**Comment by rajdip-b** [comment]
if @yetanotheraryan is not attempting it, then yes.

**Comment by vr-varad** [comment]
hey @yetanotheraryan are u working on this issue

**Comment by bluetile** [comment]
I'm working on this.

**Comment by rajdip-b** [comment]
Can you mark your attempt using `/attempt`?

**Comment by rajdip-b** [comment]
Yes, please do that. Ideally, we would like to import the zod schemas like `@keyshade/schema/raw`. Try playing around with the import patch config in the package if possible.

**Comment by rajdip-b** [comment]
Hey, yes we made a ton of changes to the responses. We condensed the responses as much as we could. I would recommend you to copy the types from `<module>.types.ts`. For instance, copy `HydratedSecret` from `secret.types.ts` in the api app. Additionally, you could go through the service layer (`secret.service.ts`) and see the various return types of the public functions. Let me know if that clears things up a bit.
